package/com.unity.formats.usd/Tests/Editor/USDPayloadComponentTests.cs

```diff
@@ -218,14 +218,20 @@ public IEnumerator ChangingUsdPayloadStateFromUnloadedToLoaded_MarksSceneDirty()
             Assert.IsFalse(m_UnityScene.isDirty, "Scene should not be dirty after after saving");
 
             // Change payload state and wait one frame to ensure Update runs
-            m_usdRoot.GetComponentInChildren<UsdPayload>().Load();
-            yield return null;
+            var payloadRoot = m_usdRoot.GetComponentInChildren<UsdPayload>();
+            var payloadRootChildCount = payloadRoot.transform.childCount;
+
+            payloadRoot.Load();
+            yield return new WaitUntil(() => m_UnityScene.isDirty == true);
+
+            Assert.IsTrue(payloadRoot == null, "Payload Load should reset USD related components and older references to it should be null");
 
-            Assert.IsTrue(m_UnityScene.isDirty, "Scene should be dirty after changing UsdPayload objects");
+            var newPayloadRoot = m_usdRoot.GetComponentInChildren<UsdPayload>();
+            Assert.AreNotEqual(payloadRootChildCount, newPayloadRoot.transform.childCount, "Payload Load should generate the payload objects");
```

Comment:
I think the addition of checking that the payload load has actually changed the hierarchy is good, as I think that is behaviour we don't check (at least that I can see in this file!). Could you please break that out into a separate test- as long as there isn't already a test doing that in a different file :) 

I believe the inputs to the Assert need to be switched, as the 'expected' to be compared against would be the newPayloadRoot, with the 'actual' being the payloadRootChildCount. Small thing, but makes diagnosing a test failure easier.